parachain/pallets/basic-channel/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+//! # Basic Channel
+//!
+//! The Basic Channel module is is a non-incentivized bridge between Ethereum and Pokadot
+//! ecosystems. It is meant to be a low-barrier entry to cross-chain communication without
+//! guarantees, but also without fees.
+//!
+//! ## Implementation
+//!
+//! Before a [`Message`] is dispatched to a target [`Application`], it is submitted to a [`Verifier`] for verification.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Calls
+//!
+//! - `submit`: Submit a message for verification and dispatch.
+//!
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult};
+use frame_system::{self as system, ensure_signed};
+
+use artemis_core::{
+	Message, MessageCommitment, MessageDispatch, MessageId, SubmitOutbound, Verifier,
+};
+use channel::{inbound::BasicInboundChannel, outbound::BasicOutboundChannel};
+use envelope::Envelope;
+use primitives::{InboundChannel, InboundChannelData, OutboundChannel, OutboundChannelData};
+use sp_core::H160 as EthAddress;
+use sp_std::convert::TryFrom;
+use sp_std::prelude::*;
+
+mod channel;
+mod envelope;
+#[cfg(test)]
+mod mock;
+mod primitives;
+#[cfg(test)]
+mod tests;
+
+type MessageNonce = u64;
+
+pub trait Config: system::Config {
+	type Event: From<Event<Self>> + Into<<Self as system::Config>::Event>;
+
+	/// Verifier module for message verification.
+	type Verifier: Verifier;
+
+	/// Used by outbound channels to persist messages for outbound delivery.
+	type MessageCommitment: MessageCommitment;
+
+	/// Verifier module for message verification.
+	type MessageDispatch: MessageDispatch<MessageId>;
+}
+
+decl_storage! {
+	trait Store for Module<T: Config> as BasicChannelModule {
+		/// Storage for inbound channels.
+		pub InboundChannels: map hasher(identity) EthAddress => InboundChannelData;
```

Comment:
By convention we just use `H160` to represent ethereum addresses, without using an alias like `EthAddress`.